be/src/common/status.cpp

```diff
@@ -83,6 +84,37 @@ Status::Status(const StatusPB& s) : _state(nullptr) {
 
 Status::Status(TStatusCode::type code, Slice msg, Slice ctx) : _state(assemble_state(code, msg, ctx)) {}
 
+void Status::access_directory_of_inject() {
+    std::string directs = starrocks::config::directory_of_inject;
+    vector<string> fields = strings::Split(directs, ",");
+    for (const auto& direct : fields) {
+        dircetory_enable[direct] = true;
+    }
+}
+
+// direct_name is like "../src/exec/pipeline" and so on.
+bool Status::in_directory_of_inject(const std::string& direct_name) {
+    if (dircetory_enable.empty()) {
+        return false;
+    }
+
+    vector<string> fields = strings::Split(direct_name, "/");
+    if (fields.size() > 1) {
+        std::stringstream ss;
+        for (int i = 1; i < fields.size(); ++i) {
+            ss << "/" << fields[i];
+            if (dircetory_enable[ss.str()]) {
```

Comment:
Maybe use `dircetory_enable.find(ss.tr()) != dircetory_enable.end()`?
Otherwise, the not existed key will be created, when calling `dircetory_enable[key]`/

Reply:
done.